.github/workflows/auto-publish-crates-upon-release.yml

```
@@ -0,0 +1,19 @@
+name: Publish-Crate-Upon-Release
+on:
+  release:
+    types: [published]
+
+jobs:
+  publish-automatically:
+    runs-on: ubuntu-latest
+    steps:
+      -  uses: actions/checkout@v2
+      -  uses: actions-rs/toolchain@v1
+         with:
+           toolchain: stable
+           override: true
+
+      - name: publish crates
+        uses: katyo/publish-crates@v1
+        with: 
+          registry-token: ${{ secrets.CARGO_API_TOKEN }}
```

This is a matter of personal taste, but I'm a bit "paranoid" about depending on external GitHub actions. Especially if they don't have much adoption or if they are doing something "trivial".

I've looked at the features provided by this extension and I couldn't find something that we need and is not covered by `cargo publish`. IMHO `cargo publish` is already doing the basic checks we would need.

I would start with something like that, and then eventually migrate to this action or to something else if we need more features:

```suggestion
      - name: publish crates
        run: cargo publish --token ${{ secrets.CARGO_API_TOKEN }}
```

---

That' a fair point. We also need to think about whose token we use, as its tied to a GitHub oauth2 grant. I am happy to self-nominate my account, but I also think we should include other sigstore-rs codeowners (aka yourself @flavio)

---

I added a token for now under `CARGO_API_TOKEN`, we can try that out @cburkhardt27

---

A common pattern I've seen on crates.io is the following one:

  * Create a bot account on crates-io: add it as one of the owners of the crate
  * Add the human maintainers of the library as maintainers too

We might want to do that... I don't know if the Sigstore organization has already a bot account that is used for other purposes. Otherwise we could simply create a new bot account on crates.io

---

OK. We now have a sigstore-bot account on crates.io: https://crates.io/users/sigstore-bot

The bot's token is added to `CARGO_API_TOKEN`

@cburkhardt27 , you should be able to make the changes flavio recommends.

@flavio Let us know when the oci-distribution PR merges that we need before we can make the first release.

---

Thanks for the feedback! I updated the PR to `cargo publish`